compiler/rustc_ast/src/ast.rs

```diff
@@ -2554,10 +2554,9 @@ pub enum AttrStyle {
 }
 
 rustc_index::newtype_index! {
-    pub struct AttrId {
-        ENCODABLE = custom
-        DEBUG_FORMAT = "AttrId({})"
-    }
+    #[custom_encodable]
+    #[debug_format = "AttrId({})]"]
+    pub struct AttrId {}
```

Comment:
Why not use a unit struct? 
```rs
pub struct AttrId;
```

Reply:
It's not really a unit struct, it's just that the field will be created by the macro

Reply:
We have `const`s inside the braces inside some invocations so for consistency (and ease of implementation for the macro) it makes sense to always require them.

Reply:
Thanks for the clarifying!